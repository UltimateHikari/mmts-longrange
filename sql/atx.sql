create table atx_test1(a text);


-- check that commit of autonomous tx will not steal locks from parent tx
begin;
    insert into atx_test1 values (1);
    select count(*) from pg_locks where transactionid=txid_current();
    begin autonomous;
        insert into atx_test1 values (1);
        select count(*) from pg_locks where transactionid=txid_current();
    commit;
    -- here we still should see our lock
    select count(*) from pg_locks where transactionid=txid_current();
commit;

drop table atx_test1;
