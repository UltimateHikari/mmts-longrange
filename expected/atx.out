create table atx_test1(a text);
-- check that commit of autonomous tx will not steal locks from parent tx
begin;
    insert into atx_test1 values (1);
    select count(*) from pg_locks where transactionid=txid_current();
 count 
-------
     1
(1 row)

    begin autonomous;
        insert into atx_test1 values (1);
        select count(*) from pg_locks where transactionid=txid_current();
 count 
-------
     1
(1 row)

    commit;
    -- here we still should see our lock
    select count(*) from pg_locks where transactionid=txid_current();
 count 
-------
     1
(1 row)

commit;
drop table atx_test1;